x-pack/plugins/apm/server/lib/services/get_services/get_services_items.ts

```diff
@@ -3,88 +3,80 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-
-import { mergeProjection } from '../../../../common/projections/util/merge_projection';
-import {
-  PROCESSOR_EVENT,
-  AGENT_NAME,
-  SERVICE_ENVIRONMENT,
-  TRANSACTION_DURATION,
-} from '../../../../common/elasticsearch_fieldnames';
+import { uniq } from 'lodash';
+import { arrayUnionToCallable } from '../../../../common/utils/array_union_to_callable';
 import { PromiseReturnType } from '../../../../typings/common';
 import {
   Setup,
   SetupTimeRange,
   SetupUIFilters,
 } from '../../helpers/setup_request';
 import { getServicesProjection } from '../../../../common/projections/services';
+import {
+  getTransactionDurationAvg,
+  getAgentName,
+  getTransactionRate,
+  getErrorRate,
+  getEnvironments,
+} from './get_metrics';
 
 export type ServiceListAPIResponse = PromiseReturnType<typeof getServicesItems>;
-export async function getServicesItems(
-  setup: Setup & SetupTimeRange & SetupUIFilters
-) {
-  const { start, end, client } = setup;
-
-  const projection = getServicesProjection({ setup });
-
-  const params = mergeProjection(projection, {
-    body: {
-      size: 0,
-      aggs: {
-        services: {
-          terms: {
-            ...projection.body.aggs.services.terms,
-            size: 500,
-          },
-          aggs: {
-            avg: {
-              avg: { field: TRANSACTION_DURATION },
-            },
-            agents: {
-              terms: { field: AGENT_NAME, size: 1 },
-            },
-            events: {
-              terms: { field: PROCESSOR_EVENT },
-            },
-            environments: {
-              terms: { field: SERVICE_ENVIRONMENT },
-            },
-          },
-        },
-      },
-    },
-  });
-
-  const resp = await client.search(params);
-  const aggs = resp.aggregations;
-
-  const serviceBuckets = aggs?.services.buckets || [];
-
-  const items = serviceBuckets.map((bucket) => {
-    const eventTypes = bucket.events.buckets;
+export type ServicesItemsSetup = Setup & SetupTimeRange & SetupUIFilters;
+export type ServicesItemsProjection = ReturnType<typeof getServicesProjection>;
 
-    const transactions = eventTypes.find((e) => e.key === 'transaction');
-    const totalTransactions = transactions?.doc_count || 0;
+export async function getServicesItems(setup: ServicesItemsSetup) {
+  const params = {
+    projection: getServicesProjection({ setup, noEvents: true }),
+    setup,
+  };
 
-    const errors = eventTypes.find((e) => e.key === 'error');
-    const totalErrors = errors?.doc_count || 0;
+  const [
+    transactionDurationAvg,
+    agentName,
+    transactionRate,
+    errorRate,
+    environments,
+  ] = await Promise.all([
+    getTransactionDurationAvg(params),
+    getAgentName(params),
+    getTransactionRate(params),
```

Comment:
Since we are retrieving more than one transaction rate
```suggestion
    getTransactionRates(params),
```

Reply:
Done